File: 110.cpp
class Solution {
public:
    bool isBalanced(TreeNode* root) {
        if (!root) return true;
        if (abs(depth(root->left) - depth(root->right)) > 1){
            return false;
        }
        return isBalanced(root->left) && isBalanced(root->right);
    }

    
    int depth(TreeNode* root){
        if(!root){
            return 0;
        }

        int leftHeight = depth(root->left);
        int rightHeight = depth(root->right);
        int heightMiddle = 1 + max(leftHeight, rightHeight);
        return heightMiddle;

    }
};

File: 110.java
class Solution {
    public boolean isBalanced(TreeNode root){
        return recur(root) != -1;
    }

    public int recur(TreeNode root){
        if (root == null) return 0;
        int leftTree = recur(root.left);
        if (leftTree == -1) return -1;
        int rightTree = recur(root.right);
        if (rightTree == -1) return -1;
        return Math.abs(leftTree - rightTree) < 2? Math.max(leftTree, rightTree)+1:-1;
    } 
};